### PR TITLE
Async PARL_IO support

### PR DESCRIPTION
**Pull request by bjoernQ**
This adds async support for the PARL_IO peripheral

### CODE REVIEW

File: esp-hal-common/src/parl_io.rs [hunk 1]
```diff
@@ -1924,5 +2086,49 @@ mod private {
 
             reg_block.rx_genrl_cfg.read().rx_eof_gen_sel().bit_is_set()
         }
+
+        #[cfg(feature = "async")]
```
Comment (MabezDev) [inline esp-hal-common/src/parl_io.rs:2090]: Any reason to cfg these behind the async flag? They seem useful outside of async too.

Comment (bjoernQ) [reply esp-hal-common/src/parl_io.rs:2090]: that was just to mute warnings in the non-async case - was considering to use allow-unused instead 🤔

Comment (MabezDev) [reply esp-hal-common/src/parl_io.rs:2090]: Oh, sorry I thought these are impl on the parlio driver itself, I now see they're private so we can leave them as is I think :)

**Review by MabezDev — APPROVED** [review]
Looks good to me!

**Pull request merged**